File: app/src/main/java/com/kaiserpudding/howtheywrite/database/CharacterDao.java
package com.kaiserpudding.howtheywrite.database;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;
import com.kaiserpudding.howtheywrite.model.Character;
import com.kaiserpudding.howtheywrite.model.Lesson;
import java.util.List;

@Dao
public interface CharacterDao {

  @Insert(onConflict = OnConflictStrategy.REPLACE)
  long insertCharacter(Character characterWord);

  @Delete
  void deleteCharacter(Character characterWord);

  @Update
  void updateCharacter(Character characterWord);

  @Query("SELECT * FROM characters WHERE id = :id LIMIT 1")
  Character getCharacterById(int id);

  @Query("SELECT * FROM characters WHERE id IN (:ids)")
  List<Character> getCharactersByIds(int[] ids);

  @Query("SELECT * FROM characters")
  LiveData<List<Character>> getAllCharacters();

}

File: HowTheyWrite_android/app/src/main/java/com/kaiserpudding/howtheywrite/database/CharacterDao.kt
package com.kaiserpudding.howtheywrite.database

import android.arch.lifecycle.LiveData
import android.arch.persistence.room.*
import com.kaiserpudding.howtheywrite.model.Character

@Dao
interface CharacterDao {

    @get:Query("SELECT * FROM characters")
    val allCharacters: LiveData<List<Character>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertCharacter(characterWord: Character): Long

    @Delete
    fun deleteCharacter(characterWord: Character)

    @Update
    fun updateCharacter(characterWord: Character)

    @Query("SELECT * FROM characters WHERE id = :id LIMIT 1")
    fun getCharacterById(id: Int): Character

    @Query("SELECT * FROM characters WHERE id IN (:ids)")
    fun getCharactersByIds(ids: IntArray): Array<Character>
}
